VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp

```diff
@@ -241,8 +241,91 @@ TArray<FInitialShapeFace> CreateDefaultInitialFaces()
 	return InitialFaces;
 }
 
+bool IsCongruentToDefaultInitialShape(const TArray<FInitialShapeFace>& InitialFaces)
+{
+	const TArray<FInitialShapeFace> DefaultInitialFaces = CreateDefaultInitialFaces();
+	check(DefaultInitialFaces.Num() == 1);
+	const TArray<FVector> DefaultVertices = DefaultInitialFaces[0].Vertices;
+	check(DefaultVertices.Num() == 4);
+
+	if (InitialFaces.Num() == DefaultInitialFaces.Num())
+	{
+		const TArray<FVector> Vertices = InitialFaces[0].Vertices;
+
+		if(Vertices.Num() == DefaultVertices.Num())
+		{
+			const FVector FirstVertex = DefaultVertices[0];
+			int32 InitialIndexOffset;
+			if(Vertices.Find(FirstVertex,InitialIndexOffset))
+			{
+				bool bIsDefaultInitialShape = true;
+				int32 IndexOffset = 0;
+				for (int32 CurrentIndex = 0; CurrentIndex < DefaultVertices.Num(); CurrentIndex++)
+				{
+					if (!Vertices[(InitialIndexOffset + CurrentIndex) % Vertices.Num()].Equals(DefaultVertices[CurrentIndex]))
+					{
+						bIsDefaultInitialShape = false;

```

Comment:
I guess we could early return false here and return true below and get rid of `bIsDefaultInitialShape`.

Reply:
changed accordingly